examples/glium_gl_area/main.rs

```diff
@@ -0,0 +1,280 @@
+                    varying vec3 vColor;
+                    void main() {
+                        gl_Position = vec4(position, 0.0, 1.0) * matrix;
+                        vColor = color;
+                    }
+                ",
+
+                fragment: "
+                    #version 110
+                    varying vec3 vColor;
+                    void main() {
+                        gl_FragColor = vec4(vColor, 1.0);
+                    }
+                ",
+            },
+
+            100 => {
+                vertex: "
+                    #version 100
+                    uniform lowp mat4 matrix;
+                    attribute lowp vec2 position;
+                    attribute lowp vec3 color;
+                    varying lowp vec3 vColor;
+                    void main() {
+                        gl_Position = vec4(position, 0.0, 1.0) * matrix;
+                        vColor = color;
+                    }
+                ",
+
+                fragment: "
+                    #version 100
+                    varying lowp vec3 vColor;
+                    void main() {
+                        gl_FragColor = vec4(vColor, 1.0);
+                    }
+                ",
+            },
+        )
+        .unwrap();
+
+        Some(Renderer {
+            context,
+            vertex_buffer,
+            index_buffer,
+            program,
+        })
+    }
+
+    fn draw(&self) {
+        let mut frame = self.context.draw();
+
+        let uniforms = uniform! {
+            matrix: [
+                [1., 0., 0., 0.],
+                [0., 1., 0., 0.],
+                [0., 0., 1., 0.],
+                [0., 0., 0., 1f32]
+            ]
+        };
+
+        frame.clear_color(0., 0., 0., 0.);
+        frame
+            .draw(
+                &self.vertex_buffer,
+                &self.index_buffer,
+                &self.program,
+                &uniforms,
+                &Default::default(),
+            )
+            .unwrap();
+        frame.finish().unwrap();
+    }
+}
+
+fn main() {
+    // Load GL pointers from epoxy (GL context management library used by GTK)
+    {
+        #[cfg(unix)]
+        let library = libloading::os::unix::Library::this();
+        #[cfg(windows)]
+        let library =
+            libloading::os::windows::Library::open_already_loaded("libepoxy-0.dll").unwrap();
+
+        epoxy::load_with(|name| {
+            unsafe { library.get::<_>(name.as_bytes()) }
+                .map(|symbol| *symbol)
+                .unwrap_or(ptr::null())
+        });
+    }
+
+    let application = gtk::Application::new(
+        Some("com.github.gtk-rs.examples.glium-gl-area"),
+        Default::default(),
+    );
+    application.connect_activate(build_ui);
+    application.run();
+}
+
+fn build_ui(application: &gtk::Application) {
+    let window = gtk::ApplicationWindow::new(application);
+
+    window.set_title(Some("Glium in GLArea"));
+
+    let widget = gtk::GLArea::new();
```

Comment:
You could have made it a GLArea subclass but no strong opinions on that

Reply:
A subclass would make conceptually more sense IMHO

Reply:
Especially you wouldn't need the weak ref anymore then